packages/core/src/react-integration/newhooks/useDle.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import type {
+  Denormalize,
+  DenormalizeNullable,
+  ErrorTypes,
+  EndpointInterface,
+  FetchFunction,
+  Schema,
+} from '@rest-hooks/endpoint';
+import useController from '@rest-hooks/core/react-integration/hooks/useController';
+import { ExpiryStatus } from '@rest-hooks/endpoint';
+import { StateContext } from '@rest-hooks/core/react-integration/context';
+import { useContext, useMemo } from 'react';
+
+type CondNull<P, A, B> = P extends null ? A : B;
+
+type StatefulReturn<S extends Schema | undefined, P> = CondNull<
+  P,
+  {
+    data: DenormalizeNullable<S>;
+    loading: false;
+    error: undefined;
+  },
+  | {
+      data: Denormalize<S>;
+      loading: false;
+      error: undefined;
+    }
+  | { data: DenormalizeNullable<S>; loading: true; error: undefined }
+  | { data: DenormalizeNullable<S>; loading: false; error: ErrorTypes }
+>;
+
+/**
+ * Use async date with { data, loading, error } (DLE)
+ * @see https://resthooks.io/docs/guides/no-suspense
+ */
+export default function useDle<
```

Comment:
How do you feel about `useDLE` instead?

Reply:
I was thinking about that. It does appear more pleasing to the eye and makes it more obvious it's a TLA

Reply:
Initialisms should always be full caps; Java-style camel casing algorithms are horrific